common/presentation/src/main/kotlin/com/pandulapeter/myTodos/presentation/ui/Demo2.kt

```diff
@@ -18,80 +20,66 @@ import com.pandulapeter.myTodos.model.Issue
 import com.pandulapeter.myTodos.presentation.resources.Dimension
 import com.pandulapeter.myTodos.presentation.resources.Text
 import com.pandulapeter.myTodos.presentation.utilities.get
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.launch
 
 @Composable
-internal fun ColumnScope.Demo2(
-    getIssues: GetIssuesUseCase = get()
+internal fun Demo2(
+    getIssues: GetIssuesUseCase = get(),
 ) {
     var issues by remember { mutableStateOf<ResultWrapper<List<Issue>>?>(null) }
+    var retry by remember { mutableStateOf(Any()) }
 
-    fun loadIssues() {
-        issues = null
-        GlobalScope.launch {
-            val newIssues = getIssues()
-            launch(Dispatchers.Main) {
-                issues = newIssues
-            }
-        }
+    LaunchedEffect(getIssues, retry) {
+        issues = getIssues()
     }
 
-    when (val currentIssues = issues) {
-        null -> {
-            LoadingIndicator()
-            loadIssues()
-        }
-        is ResultWrapper.Error -> {
-            ErrorState(
-                onTryAgainButtonPressed = ::loadIssues
+    Column(
+        modifier = Modifier.fillMaxWidth(),
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        when (val currentIssues = issues) {
+            null -> LoadingIndicator()
+            is ResultWrapper.Error -> ErrorState(
+                onTryAgainButtonPressed = { issues = null; retry = Any() }
```

Comment:
To trigger the re-launching of Launched effect I'm sending a different instance of Any() which acts as an event, but this event should really be sent to a ViewModel.